scripts/configurator/parser/cmake_row.py

```diff
@@ -0,0 +1,105 @@
+import re
+from typing import Match, Optional
+from enum import Enum, auto
+
+class row_type(Enum):
+   EMPTY = auto()
+   SLASH_COMMENT = auto()
+   POUND_COMMENT = auto()
+   VARIABLE = auto()
+   NOT_PARSABLE = auto()
+
+class configurator_type(Enum):
+   BOOL = auto()
+   FILEPATH = auto()
+   STRING = auto()
+   INTERNAL = auto()
+
+class cmake_row:
+   """
+   simple utility class that abstracts away some details from the parsing
+   process from a single CMakeCache.txt row.
+   """
+   regex_expression = re.compile(r"^([A-Za-z_0-9]+?):([A-Za-z]+?)=(.*)$")
+
+   NORMALIZED_BOOLS = {
+      "ON" : True,
+      "OFF": False,
+      "1" : True,
+      "0" : False,
+      "TRUE": True,
+      "FALSE": False,
+   }
+
+   string_to_conf_type  = {
+      "BOOL": configurator_type.BOOL,
+      "FILEPATH": configurator_type.FILEPATH,
+      "STRING": configurator_type.STRING,
+      "INTERNAL": configurator_type.INTERNAL,
+   }
+
+   def __init__(self, raw_row: str, row_number: int, comment: str = ""):
+      """
+      strips variable and assigns its correct type by matching the regex
+      and verifying if it's a comment.
+      """
+      row  = raw_row.strip()
+      matches: Optional[Match[str]]  = re.match(self.regex_expression, row)
+      self.row_type = row_type.NOT_PARSABLE
+      self.val: str | bool = ""
+      self.name: str = ""
+      self._cmake_type_str: str # used to serialize back the value
+      self.configurator_type: configurator_type
+      self.row_number: int = row_number
+      self.comment: str = comment
+
+      if not len(row):
+         self.row_type = row_type.EMPTY
+         return
+
+      if self.is_comment(row):
+         if self.is_slash_comment(row):
+            self.row_type = row_type.SLASH_COMMENT
+            self.val = row[2:]
+            return
+
+         self.row_type = row_type.POUND_COMMENT
+         self.val = row[1:]
+         return
+
+      elif matches: # this guarantees that the regex has exactly 3 matches
+         groups = matches.groups()
+         self.name = groups[0]
+         self.row_type = row_type.VARIABLE
+         self.val = groups[2]
+         self._cmake_type_str = groups[1]
+         self.configurator_type = self.string_to_conf_type[self._cmake_type_str]
+         if self.configurator_type == configurator_type.BOOL:
```

Comment:
That's the place where you should create a cmake_variable can give it the string-representation in order to the variable class to de-serialize it, instead of doing it everything in the `cmake_row` class.